backend/compact-connect/lambdas/python/common/cc_common/data_model/user_client.py

```
@@ -343,3 +348,45 @@ def delete_user(self, *, compact: str, user_id: str) -> None:
             if e.response['Error']['Code'] == 'ConditionalCheckFailedException':
                 raise CCNotFoundException('User not found') from e
         return
+
+    def reinvite_user(self, *, email: str) -> None:
+        """
+        Reinvite a staff user by resetting their password and resending their invite email
+        :param str email: The user's email address
+        """
+        try:
+            # Check their current status
+            user_data = self.config.cognito_client.admin_get_user(
+                UserPoolId=self.config.user_pool_id,
+                Username=email,
+            )
+
+            # If they're in CONFIRMED state, we need to reset their password first
+            if user_data['UserStatus'] == 'CONFIRMED':
```

So this method is only accounting for 'CONFIRMED' or 'FORCE_RESET_PASSWORD' status values, but there are other potential values right? Shouldn't we be checking for those as well? If it is any other status the resend api call will fail

---

@landonshumway-ia , I've revised the logic in this method to be more explicit about checking user status:
- If user has `CONFIRMED`, reset their password before reinviting;
- If user has `FORCE_CHANGE_PASSWORD` or `RESET_REQUIRED` state, resend invite;
- If user has any other state, raise an exception as other states are unexpected.

---

@jusdino I'm not sure about the `REST_REQUIRED` state. As I recall you can only resend the invite using the AdminCreateUser when the cognito user is in the `FORCE_CHANGE_PASSWORD` state, so I believe the API call will fail if the user is in a `REST_REQUIRED` state. BTW, the only way the user is put in this state is if they the user's password is deactivated through the console or using the 'AdminResetUserPassword' api, so I think we can assume they should never end in that state either, and we should just raise an exception.

---

jk, looks like it works 😆

---

No, you were right. My test was bad. Tweaking again.